### PR TITLE
Adapt CI so it covers both new and old Macs, and installs required additional dependencies on M1

### PR DESCRIPTION
**Pull request by alessandrofelder**
## Description

**What is this PR**

- [x] Bug fix
- [ ] Addition of a new feature
- [ ] Other

**Why is this PR needed?**

`macos-latest` is now a Silicon Mac runner, so we need to adapt CI accordingly

**What does this PR do?**
* ensures CI is run on both Mac chip architectures
* runs tensorflow include guard tests on Silicon mac instead.

## References

Partial solution to https://github.com/brainglobe/BrainGlobe/issues/63

## How has this PR been tested?

CI that was breaking before, now passes

## Is this a breaking change?

Now

## Does this PR require an update to the documentation?

No

## Checklist:

- [n/a] The code has been tested locally
- [n/a] Tests have been added to cover all new functionality (unit & integration)
- [n/a] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: .github/workflows/test_and_deploy.yml [hunk 1]
```diff
@@ -42,12 +42,14 @@ jobs:
...
-        - os: macos-latest
-          python-version: "3.10"
-        - os: windows-latest
-          python-version: "3.10"
+          - os: macos-13
+            python-version: "3.10"
+          - os: macos-latest
+            python-version: "3.10"
+          - os: windows-latest
+            python-version: "3.10"
```
Comment (sfmig) [inline .github/workflows/test_and_deploy.yml:52]: question: do you know why we don't need to install `hdf5` here?

**Review by sfmig — APPROVED** [review]
thanks @alessandrofelder, looks good too! 😁 Just a couple of questions out of curiosity.

File: .github/workflows/test_and_deploy.yml (hunk 1 above)
Comment (alessandrofelder) [reply .github/workflows/test_and_deploy.yml:52]: The `hdf5` installation issue on M1 is linked to depending on `pandas`, which cellfinder doesn't.

File: .github/workflows/test_include_guard.yaml [hunk 2]
```diff
@@ -35,7 +35,12 @@ jobs:
             import cellfinder.core
             import cellfinder.napari
 
-      - name: Uninstall tensorflow
+      - name: Uninstall tensorflow-macos on Mac M1
+        if: matrix.os == 'macos-latest'
```
[the remark this answers is not in the document]
Comment (alessandrofelder) [reply .github/workflows/test_include_guard.yaml:39]: We install a [different `tensorflow` depending on ARM and Intel architectures](https://github.com/brainglobe/cellfinder/blob/0fd0a5e3712047b5cac138f329a3c70dd48bca66/pyproject.toml#L33), and I wanted to cover both with the same number of runs as before?

**Pull request merged**